tests/unit/core/sagas/test_autocomplete.js

```diff
@@ -0,0 +1,83 @@
+import SagaTester from 'redux-saga-tester';
+
+import autocompleteReducer, {
+  autocompleteStart,
+  AUTOCOMPLETE_LOADED,
+} from 'core/reducers/autocomplete';
+import { CLEAR_ERROR } from 'core/constants';
+import * as api from 'core/api';
+import apiReducer from 'core/reducers/api';
+import autocompleteSaga from 'core/sagas/autocomplete';
+import {
+  createFakeAutocompleteResult,
+  dispatchSignInActions,
+} from 'tests/unit/amo/helpers';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockApi = sinon.mock(api);
+    const initialState = dispatchSignInActions().state;
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: {
+        api: apiReducer,
+        autocomplete: autocompleteReducer,
+      },
+    });
+    sagaTester.start(autocompleteSaga);
+  });
+
+  function _autocompleteStart(params) {
+    sagaTester.dispatch(autocompleteStart({
+      errorHandlerId: 'create-stub-error-handler-id',
+      ...params,
+    }));
+  }
+
+  it('calls the API for suggestions', async () => {
+    const results = [createFakeAutocompleteResult({ name: 'first-result' })];
+    const filters = { query: 'test' };
+
+    mockApi
+      .expects('autocomplete')
+      .once()
+      .returns(Promise.resolve({ results }));
+
+    _autocompleteStart({ filters });
+
+    // The saga should respond by dispatching the autocomplete loaded action.
+    await sagaTester.waitFor(AUTOCOMPLETE_LOADED);
```

Comment:
I know you copied this from the last test (d'oh) but I think using action creators would make it a bit more robust:

```js
const expectedLoadAction = autocompleteLoad(results);

await sagaTester.waitFor(expectedLoadAction.type);
mockApi.verify();

const loadAction = sagaTester.getCalledActions()[2];
expect(loadAction).toEqual(expectedLoadAction);
```